src/stages/evaluate.ts

```diff
@@ -227,34 +234,31 @@ function findNextSlotToPrompt<T, G>(getState: () => WolfState, abilities: Abilit
     return null
   }
 
-  const unfilledSlots = getUnfilledSlots(getState, abilities, focusedAbility)
+  const unfilledSlots = getUnfilledSlots(getState, slots, abilities, focusedAbility)
   log('enabled and unfilled slots are %o', unfilledSlots.map(_ => _.name))
   if (unfilledSlots.length === 0) {
     log('no enabled slots found')
     log('exiting findNextSlotToPrompt()')
     return null // no slots need to be filled in current focused ability
   }
 
-  // REMAINING SLOTS NEED TO BE FILLED
-  // sort slots by order value
-  const sortedSlots = unfilledSlots.sort((a, b) => {
-    if (!a.order) { a.order = 100 }
-    if (!b.order) { b.order = 100 }
-    return a.order - b.order
-  })
-  log('sorted Slots are %o', sortedSlots.map(_ => _.name))
-  log('found %s slot.. returning..', sortedSlots[0].name)
+  log('sorted Slots are %o', unfilledSlots.map(_ => _.name))
+  log('found %s slot.. returning..', unfilledSlots[0].name)
   log('exiting findNextSlotToPrompt()')
   return {
-    slotName: sortedSlots[0].name,
+    slotName: unfilledSlots[0].name,
     abilityName: focusedAbility
   }
 }
 
 /**
  * Check if there are any abilities with all enabled slots filled.
  */
-function getAbilitiesCompleted<T, G>(getState: () => WolfState, abilities: Ability<T, G>[]): string[] {
+function getAbilitiesCompleted<T, G>(
+  getState: () => WolfState,
+  slots: Slot<G>[],
+  abilities: Ability<T, G>[]
```

Comment:
Parameter `slots` and `abilities` can be simplified to pass in `flow` which contains both. Simplifies parameter from 2 -> 1